pygfx/utils/transform.py

```diff
@@ -524,6 +524,17 @@ class AffineTransform(AffineBase):
     is_camera_space : bool
         If True, the transform represents a camera space which means that it's
         ``forward`` and ``right`` directions are inverted.
+    state_basis : str
+        One of "components" or "matrix". If "components", the transform will
+        store its state in the form of position, rotation, and scale. If "matrix",
+        the transform will store its state in the form of a matrix. The default
+        is "components". This can have performance implications when the
+        transform is used in a context where the state is updated in one
+        form but queried in another.
+    matrix : ndarray, [4, 4]
+        The affine matrix describing this transform. If state_basis is set to
+        "matrix", this matrix will be used as the basis for the transform's
+        state.
```

Comment:
And if `state_basis` is "components", it would also override the `position`etc, right?

Reply:
Well, no. If state_basis is components AND you provide a matrix in the constructor, it will effectively be ignored.

Reply:
I also tried to clarify this in the docstrings.